ansible/roles/test/tasks/buffer_profile.yml

```diff
@@ -0,0 +1,30 @@
+#
+# ansible-playbook test_sonic.yml -i inventory --limit arc-switch1028 --tags buffer_profile -e skip_interface_check=true
+#
+#
+
+- set_fact:
+    port_name="Ethernet64"
+    minigr="[ 'minigraph0.xml', 'minigraph1.xml', 'minigraph2.xml']"
+    test_files_dir="roles/test/files/buffer_profile"
+
+- block:
+    - name: Copy test files to DUT
+      copy: src={{ item }} dest=/tmp/
+      with_fileglob:
+        - "{{ test_files_dir }}/*"
+
+    - name: Generate buffers config
+      shell: sonic-cfggen -m /tmp/{{ item.1 }} -t /usr/share/sonic/templates/msn27xx.32ports.buffers.json.j2 >/tmp/buffers{{ item.0}}.json
+      with_indexed_items:
+        - "{{ minigr }}"
+
+    - name: Compare generated buffers configuration with the expected
+      shell: diff /tmp/{{ (item | basename | splitext)[0]}}.json /tmp/{{ (item | basename).split('_')[0] }}.json
```

Comment:
Do you think you can merge this step with the previous step,execute generation and comparison in the same shell command to reduce number of ssh sessions needed for the test? If you do so, you can always generate target file with a same name and compare it to the right template.

Regards,
Ying